src/tests/suite/daos_test.h

```diff
@@ -439,19 +434,26 @@ test_rmdir(const char *path, bool force)
 		if (!strcmp(ent->d_name, ".") || !strcmp(ent->d_name, ".."))
 			continue;   /* skips the dots */
 
-		sprintf(fullpath, "%s/%s", path, ent->d_name);
+		D_ASPRINTF(fullpath, "%s/%s", path, ent->d_name);
```

Comment:
You need to free it in each iteration of the loop.   D_ASPRINTF is like malloc so each call will leak memory as you have it.